automatic/freecad/update_helper.ps1

```diff
@@ -1,6 +1,39 @@
 
+function Get-HeadersIfChanged() {
+param(
+    [string]$url,
+    [string]$file,
+    [ValidateSet("portable", "stable", "dev")]
+    [string]$kind
+)
+# Check for the info file and if not create it
+  if (!(Test-Path $file)) {
+	New-Item $file -ItemType file
+  }
+# Read file when existing and not forced else info is null
+  if (($global:au_Force -ne $true) -and (Test-Path $file)) {
+    $existingETag = ( Get-Content $file -Raw | ConvertFrom-Json )
+  }  else {
+    $existingETag = @{}
+  }
+# Retrieve info from file
+  $etag = Invoke-WebRequest -Method Head -Uri $url -UseBasicParsing
+  # $etag = $etag | Foreach { $_.Headers.ETag }
+  $etag = $etag | Foreach { $_.Headers["Content-Length"] }
```

Comment:
Why use the `Content-Length` instead of the `Etag`?

It would be preferred to use the etag instead of the content length as the former should be unique whenever the executable changes, but there isn't a guarantee that the executable have changed size (likely to have a changed size yes, but it have the potential to be the same nonetheless).

If you change it to be an etag, you also do not need to roll your own helper in this way, as you can reuse one of our existing scripts in that case: https://github.com/chocolatey-community/chocolatey-packages/blob/master/scripts/Update-OnETagChanged.ps1

Example Usage: https://github.com/chocolatey-community/chocolatey-packages/blob/25a92fbab82ac08524ae6dd239d358e2b266109f/automatic/rdcman/update.ps1#L51-L53 (The Get-RemoteInformation would be different though, as you donåt need to download the executable, and there is a need to check to see if you need to create a fix version or not).

Reply:
The use of `Content-Length` instead of `ETag` was due to not all sites providing the `ETag` information.
The use of the `Update-OnETagChanged` helper requires updates if used in a multi-stream package.